### DISCUSSION TITLE
Bagaimana membuat Bar Chart dengan data seperti ini di R?

### DISCUSSION THREAD

**Discussion by nicosyuu** [teamalgoritma/community#793] [Q&A] [answered]
Halo, saya mau bertanya jika saya punya data seperti ini?
![image](https://github.com/teamalgoritma/community/assets/103729742/09e46883-95a0-41d8-a138-b8cc066489fe)
lalu saya mau membuat bar chart utk each row seperti ini... Ini saya buat di excel tapi manual for each row
![image](https://github.com/teamalgoritma/community/assets/103729742/a6575d7e-9fbc-451d-bd55-de693c7b8536)
Bagaimana saya bisa membuat secara automate di R?
Terima kasih

**Comment by arkanayudhistira** [comment]
Dear @nicosyuu,

Sebelumnya saya ingin memastikan terlebih dahulu, berdasarkan contoh yang dikirimkan, Bar Chart hanya dibuat untuk satu baris saja (baris yang namanya Alvino). Karena data yang digunakan terdiri atas 30 baris, bagaimana bentuk bar chart yang diinginkan untuk baris yang lainnya? Apakah bapak ingin membuat 30 bar chart untuk tiap baris?

**Reply by nicosyuu** [reply]
iya benar, tapi saya sudah ketemu  caranya Terima Kasih atas perhatiannya

**Comment by nicosyuu** [comment, marked as answer]
```{r}
# Assuming your data is stored in a data frame named 'df'
library(ggplot2)

for (i in 1:nrow(df)) {
  student_name <- df$NAMA[i]
  student_data <- data.frame(Subject = names(df)[-(1)], Score = round(as.numeric(df[i, -(1)]), 2))

  # Create a bar chart for the current student's scores
  p <- ggplot(data = student_data, aes(x = Subject, y = Score, label = Score)) +
    geom_bar(stat = "identity", fill = "blue") +
    geom_text(vjust = -0.5) +  # Adjust the vertical position of labels
    labs(title = student_name, x = "Subjects", y = "Average Score")

  # Print the current student's bar chart
  print(p)
}
